application/src/main/java/org/thingsboard/server/service/install/update/DefaultDataUpdateService.java

```diff
@@ -203,11 +206,121 @@ public void updateData(String fromVersion) throws Exception {
                     log.info("Skipping edge events migration");
                 }
                 break;
+            case "3.5.0":
+                log.info("Updating data from version 3.5.0 to 3.5.1 ...");
+                log.info("Starting enrichment rule nodes update ...");
+                updateEnrichmentRuleNodes();
+                log.info("Finished enrichment rule nodes update!");
+                break;
             default:
                 throw new RuntimeException("Unable to update data, unsupported fromVersion: " + fromVersion);
         }
     }
 
+    private void updateEnrichmentRuleNodes() {
+        try {
+            var ruleChainIdToTenantId = new HashMap<RuleChainId, TenantId>();
+            var allNodesToUpdate = List.of(
+                    "org.thingsboard.rule.engine.metadata.TbGetOriginatorFieldsNode",
```

Comment:
Replace with the following code. We don't need issue with the update if someone decides to move the class.
```
TbGetOriginatorFieldsNode.getClass().getName()
```